external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaRelationSuite.scala

```diff
@@ -142,6 +142,118 @@ class KafkaRelationSuite extends QueryTest with SharedSQLContext with KafkaTest
     checkAnswer(df, (0 to 30).map(_.toString).toDF)
   }
 
+  test("timestamp provided for starting and ending") {
+    val (topic, timestamps) = prepareTimestampRelatedUnitTest
+
+    // timestamp both presented: starting "first" ending "finalized"
+    verifyTimestampRelatedQueryResult({ df =>
+      val startTopicTimestamps = Map(topic -> timestamps.head)
+      val startingTimestamps = JsonUtils.topicTimestamps(startTopicTimestamps)
+
+      val endTopicTimestamps = Map(topic -> timestamps(3))
+      val endingTimestamps = JsonUtils.topicTimestamps(endTopicTimestamps)
+
+      df.option("startingOffsetsByTimestamp", startingTimestamps)
+        .option("endingOffsetsByTimestamp", endingTimestamps)
+    }, topic, 0 to 20)
```

Comment:
I'm not sure I get it. Could you elaborate on the case?

Reply:
Couple of tests are something like this (oversimplified):
```
producer.send(0 to 20)
...some mambo-jumbo here...
val result = consumer.pollAll(0, 20)
assert(result, (0 to 20))
```
It would be good to test something which is not `earliest` or `latest`.

Reply:
If I understand correctly, you would like to see the expected result like 5 to 15 (filtered records), not 0 to 20 (full records). If I got you correctly, I guess we could modify this UT to test it then.

Reply:
Yeah, something like this would be good.

Reply:
Addressed.